src/wallets/index.ts

```diff
@@ -260,6 +265,34 @@ export async function addWallet(
   await addExpiration();
 }
 
+/**
+ * updates password across all accounts for the user
+ *
+ * @param newPassword new password
+ * @param prevPassword previous password to verify
+ */
+export async function updatePassword(
+  newPassword: string,
+  prevPassword: string
+) {
+  if (!(await checkPassword(prevPassword))) {
+    throw new Error("Invalid password");
+  }
+  const wallets = await getWallets();
+  for (const item of wallets) {
+    const decrypedKeyfile = await decryptWallet(
+      (item as LocalWallet).keyfile,
+      prevPassword
+    );
+    const encrypted = await encryptWallet(decrypedKeyfile, newPassword);
+
```

Comment:
After you've encrypted the wallet, the decrypted instance is no longer needed, so you should immediately clear it from the runtime memory, just as a security measure. We have the `freeDecryptedWallet()` utility function for this (from: `src/wallets/encryption.ts`).

Int this case, it would look something like this:
```ts
// remove wallet from runtime memory
freeDecryptedWallet(decrypedKeyfile);
```